library/src/main/android/permissions/dispatcher/PermissionUtils.java

```diff
@@ -106,11 +105,14 @@ private static boolean hasSelfPermission(Context context, String permission) {
         }
     }
 
-    @TargetApi(Build.VERSION_CODES.M)
     private static boolean hasSelfPermissionForXiaomi(Context context, String permission) {
-        AppOpsManager appOpsManager = (AppOpsManager) context.getSystemService(Context.APP_OPS_SERVICE);
-        int checkOp = appOpsManager.checkOp(AppOpsManager.permissionToOp(permission), Process.myUid(), context.getPackageName());
-        return checkOp == AppOpsManager.MODE_ALLOWED && checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED;
+        String permissionToOp = AppOpsManagerCompat.permissionToOp(permission);
+        if (permissionToOp == null) {
+            // in case of normal permissions(e.g. INTERNET)
+            return true;
+        }
+        int noteOp = AppOpsManagerCompat.noteOp(context, permissionToOp, Process.myUid(), context.getPackageName());
```

Comment:
Good catch, actually I'm not sure how `SecurityException` is thrown either...
I suppose it's okay since nobody has complained about it yet...YAGNI:D